File: runtime/src/main/java/com/siimkinks/sqlitemagic/entity/ConnectionProvidedOperation.java
package com.siimkinks.sqlitemagic.entity;

import com.siimkinks.sqlitemagic.DbConnection;

import androidx.annotation.CheckResult;
import androidx.annotation.NonNull;

public interface ConnectionProvidedOperation<R> {
  /**
   * Configure this operation to use provided connection for database operation.
   *
   * @param connection Database connection
   * @return Operation builder
   */
  @NonNull
  @CheckResult
  R usingConnection(@NonNull DbConnection connection);
}

File: runtime/src/main/kotlin/com/siimkinks/sqlitemagic/entity/ConnectionProvidedOperation.kt
package com.siimkinks.sqlitemagic.entity

import androidx.annotation.CheckResult
import com.siimkinks.sqlitemagic.DbConnection

interface ConnectionProvidedOperation<R> {
  /**
   * Configure this operation to use provided connection for database operation.
   *
   * @param connection Database connection
   * @return Operation builder
   */
  @CheckResult
  fun usingConnection(connection: DbConnection): R
}
